DataStructures/LinkedList/CircularLinkedList/CircularLinkedList.cs

```diff
@@ -0,0 +1,173 @@
+        public void InsertAtEnd(T data)
+        {
+            var newNode = new CircularLinkedListNode<T>(data);
+            if (IsEmpty())
+            {
+                tail = newNode;
+                tail.Next = tail;
+            }
+            else
+            {
+                newNode.Next = tail!.Next;
+                tail.Next = newNode;
+                tail = newNode;
+            }
+        }
+
+        /// <summary>
+        /// Inserts a new node after a specific value in the list.
+        /// </summary>
+        /// <param name="value">The value to insert the node after.</param>
+        /// <param name="data">The data to insert into the new node.</param>
+        public void InsertAfter(T value, T data)
+        {
+            if (IsEmpty())
+            {
+                throw new InvalidOperationException("List is empty.");
+            }
+
+            var current = tail!.Next;
+            do
+            {
+                if (current!.Data!.Equals(value))
+                {
+                    var newNode = new CircularLinkedListNode<T>(data);
+                    newNode.Next = current.Next;
+                    current.Next = newNode;
+
+                    if (current == tail)
+                    {
+                        tail = newNode;
+                    }
+
+                    return;
+                }
+
+                current = current.Next;
+            }
+            while (current != tail.Next);
+        }
+
+        /// <summary>
+        /// Deletes a node with a specific value from the list.
+        /// </summary>
+        /// <param name="value">The value of the node to delete.</param>
+        public void DeleteNode(T value)
+        {
+            if (IsEmpty())
+            {
+                throw new InvalidOperationException("List is empty.");
+            }
+
+            var current = tail!.Next;
+            var previous = tail;
+
+            do
+            {
+                if (current!.Data!.Equals(value))
+                {
+                    if (current == tail && current.Next == tail)
+                    {
+                        tail = null;
+                    }
+                    else if (current == tail)
+                    {
+                        previous!.Next = tail.Next;
+                        tail = previous;
+                    }
+                    else if (current == tail.Next)
+                    {
+                        tail.Next = current.Next;
+                    }
+                    else
+                    {
+                        previous!.Next = current.Next;
+                    }
+
+                    return;
+                }
+
+                previous = current;
+                current = current.Next;
+            }
+            while (current != tail!.Next);
+        }
+
+        /// <summary>
+        /// Displays the contents of the Circular Linked List.
+        /// </summary>
+        public void Display()
```

Comment:
Please remove display method and display-related logic from tests

Reply:
Done